src/server/game/Instances/InstanceSaveMgr.cpp

```diff
@@ -516,10 +544,27 @@ void InstanceSaveManager::_ResetSave(InstanceSaveHashMap::iterator &itr)
     // do not allow UnbindInstance to automatically unload the InstanceSaves
     lock_instLists = true;
 
+    bool shouldDelete = true;
     InstanceSave::PlayerListType &pList = itr->second->m_playerList;
-    while (!pList.empty())
+    std::vector<Player*> temp; // list of expired binds that should be unbound
+    for (Player* player : pList)
     {
         Player* player = *(pList.begin());
+        if (InstancePlayerBind* bind = player->GetBoundInstance(itr->second->GetMapId(), itr->second->GetDifficulty()))
+        {
+            assert(bind->save == itr->second);
```

Comment:
Under no circumstances you should ever use assert from <cassert> header - it does NOT generate a crash log on windows.